#32708: Migrate op to new infra: nlp_concat_heads_boltz

ssundaramTT opened the pull request:
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/32708)

### Problem description
Migrating nlp_concat_heads_boltz op to new device op infra

### What's changed

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes ([link to run](https://github.com/tenstorrent/tt-metal/actions/runs/19714813564))
- [x] L2 nightly cpp tests ([link to run](https://github.com/tenstorrent/tt-metal/actions/runs/19714806921))
- [x] New/Existing tests provide coverage for changes

ssundaramTT committed:
migrated nlp_concat_heads_boltz to new device op infra

ssundaramTT committed:
Apply suggestions from code review

Co-authored-by: Vladyslav Tsilytskyi <vtsilytskyi@tenstorrent.com>

ayerofieiev-tt commented on ttnn/cpp/ttnn/operations/experimental/transformer/nlp_concat_heads_boltz/device/nlp_concat_heads_boltz_program_factory.cpp:
```diff
@@ -4,17 +4,23 @@
 
-namespace ttnn::operations::experimental::transformer {
+namespace ttnn::operations::experimental::nlp_concat_heads_boltz {
+namespace program {
```
generally I can agree with Vlad that this namespace is not essential
and the less complexity we carry the better

ssundaramTT committed:
Update ttnn/cpp/ttnn/operations/experimental/transformer/nlp_concat_heads_boltz/device/nlp_concat_heads_boltz_program_factory.cpp

Co-authored-by: Vladyslav Tsilytskyi <vtsilytskyi@tenstorrent.com>

ssundaramTT committed:
remove program namespace

ayerofieiev-tt committed:
Initialize member variables in NLPConcatHeadsBoltzSharedVariables

Initialized KernelHandle and CBHandle members in NLPConcatHeadsBoltzSharedVariables.
